goby.md

```diff
@@ -0,0 +1,507 @@
+
+### Black hole variable
+
+```ruby
+a, _ = [1, 2]
+```
+
+Single "`_`" is a special "black hole" variable (write-only).
+
+### Class variable
+
+Unsupported.
+
+### Global variable
+
+Unsupported.
+
+## Literal
+{: .-three-column}
+
+### Keyword
+
+`def`, `true`, `false`, `nil`, `if`, `elsif`, `else`, `case`, `when`, `return`, `self`, `end`, `while`, `do`, `yield`, `next`, `class`, `module`, `break`
+
+(`then` is **not** a keyword)
+
+### String literal
+
+```ruby
+"double quote"
+'single quote'
+```
+
+Double and single quotation can be used.
+
+### Symbol literal
+
+```ruby
+:symbol           # equivalent to "symbol"
+{ symbol: "value" }
+```
+
+Goby's symbol (using `:`) is just a short hand of string literal and is always `String` class (**no** `Symbol` class!).
+
+Note that symbol literal should be "`[a-zA-Z][a-zA-Z0-9_]+`" and cannot include any other letters, signs or spaces.
+
+### Numeric literal
+
+```ruby
+year   =  2018   # Integer
+offset = -42     # Integer
+PI     = 3.14    # Float
+G      = -9.8    # Float
+```
+
+### Array literal
+
+```ruby
+[1, 2, 3, "hello", :goby, { key: "value"}]
+[1, 2, [3, 4], 5, 6]
+```
+
+### Hash literal
+
+```ruby
+h = { key: "value", key2: "value2" }
+h[:key2]   #=> value2
+```
+
+Hash literal's keys should always be symbol literals. 
+
+### Range
+
+```ruby
+(1..10).each do |x|
+  puts x*x
+end
+```
+
+"`..`" within parentheses "`( )`" represents a range.
+
+### Boolean and `nil`
+
+```ruby
+true
+false
+nil
+
+!nil  #=> true
+```
+
+`true` and `false` belongs to `Boolean` class.
+
+`nil` belongs to `Null` class and treated as `false` on conditionals.
+
+Any objects except `nil` and `false` are `true` on conditionals.
+
+### Operator
+
+```ruby
++           # unary
+**          # power
+-           # unary
+* / %       # multiplication, division, modulus
++ -         # addition, subtraction
+!           # logical inversion
+> >= < <=   # inequality comparison
+== !=       # equality comparison, negative comparison
+&&          # logical AND
+||          # logical OR
++= -=       # shorthand of addition/subtraction
+```
+
+Arithmetic and logical operators (under way).
+
+```ruby
+()          # chaning priority of interpretation
+[]          # array literal
+*           # multiple assignment
+..          # range
+```
+
+Other operators (under way)
+
+## Flow control
+{: .-three-column}
+
+### Conditional
+
+```ruby
+def foo(str)
+  if str.size > 10
+    puts "too big!"
+  else
+    if str.size < 3
+      puts "too short!"
+    else
+      puts "moderate"
+    end
+  end
+end
+```
+
+### Case
+
+### Exception
+
+### Block
+
+## Thread and channel
+
+### Concurrent array
+
+### Concurrent hash
+
+### Concurrent rw rock
+
+## Native class (fundamental)
+
+### `Object` and `Class`
+
+{: .-three-column}
+
+### `String`
+
+### `Integer`
+
+### `Array`
+
+### `Hash`
+
+### `Array`
+
+### `Range`
+
+### `Method`
+
+### `Block`
+
+### `Boolean`
```

Comment:
These doesn't seem like it should be H3 headings. How about a list?

BTW: https://devhints.io/cheatsheet-styles